File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:

        prev, maxValue = 0, -inf

        for i in range(len(nums)):
            prev = prev + nums[i]
            maxValue = max(maxValue, prev)

            if prev < 0:
                prev = 0

        return maxValue

File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        
        int prev = nums[0];
        int sum = nums[0];
        int max = nums[0];

        for(int i = 1; i < nums.length; i++){
            if(prev + nums[i] > nums[i]){
                prev = nums[i] + prev;
            } else {
                prev = nums[i];
            }            
            max = Math.max(prev, max);

        }

        return max;
    }
}